### ISSUE TITLE
Fix Request: InconsistentEqualsAndHashCode is triggered on private boolean equals(String x, String y)

### ISSUE THREAD

**Issue by FrenkelS** [jborgers/PMD-jPinpoint-rules#130]
The rule pmd:InconsistentEqualsAndHashCode is triggered on the following code:

```
private boolean equals(String value1, String value2) {
    return (obj1 == null && obj2 == null) || value1 != null && value1.equalsIgnoreCase(value2);
}
```

This rule is about `public boolean equals(Object obj)` and `public int hashCode()`, so it shouldn't be triggered on a private method that is also called "equals" and that takes two arguments and returns a boolean.
